cpp/include/cudf/table/experimental/row_operators.cuh

```diff
@@ -1217,6 +1235,52 @@ struct strong_index_comparator_adapter {
 };
 // @endcond
 
+/**
+ * @brief The adapter functor for calling a row comparator with strong index types from just one
+ *        type of integer index.
+ *
+ * In order to call the underlying row comparator with strong index types, the input integer
+ * indices to the functor must be given such that one index must be non-negative and the other must
+ * be negative:
+ *  - The non-negative index will be converted to `lhs_index_type`.
+ *  - The negative index will be normalized using @ref `normalize_index` and converted to
+ *    `rhs_index_type`.
+ *
+ * @tparam Comparator A class of device row comparator with strong index types.
+ */
+template <typename Comparator>
+struct index_normalized_comparator_adapter {
+  index_normalized_comparator_adapter(Comparator&& comparator_) : comparator(std::move(comparator_))
+  {
+  }
```

Comment:
I really don't want to be building more machinery that depends on this pattern when we know we don't need it. I guess it's fine if we all know it's 100% throwaway work and won't be merged.